xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml

```diff
@@ -247,6 +389,50 @@
   #end
 #end
 ###############################
+##   DISPLAY SHORTCUT PREFERENCE
+###############################
+#macro (displayShortcut $fieldName $label $translationKey)
+  #set ($hintKey = "XWiki.UserPreferenceShortcutsClass_${fieldName}.hint")
+  #set ($escapedSelectId = $escapetool.xml("XWiki.UserPreferenceShortcutsClass_${fieldNumber}"))
+  #set ($escapedHintId = $escapedSelectId + "_hint")
+  #set ($defaultValue = $escapetool.xml($services.localization.render($translationKey)))
+  &lt;dt&gt;
+    &lt;label #if($isEdit)for="$escapedSelectId"#end&gt;$!escapetool.xml($services.localization.render($label))&lt;/label&gt;
+    #if ($services.localization.get($hintKey))
+      &lt;span id="$escapedHintId" class='xHint'&gt;$!escapetool.xml($services.localization.render($hintKey)) ##
+      $escapetool.xml($services.localization.render('userprofile.default')): $defaultValue##
+      &lt;/span&gt;
+    #end
+  &lt;/dt&gt;
+  &lt;dd data-user-property="$!escapetool.xml($fieldName)"&gt;
+    #set ($shortcutPreference = '')
+    #set ($shortcutPreference = $doc.getObject('XWiki.UserPreferenceShortcutsClass', 'translationKey', $translationKey))
+    #set ($value = $doc.getValue('value', $shortcutPreference))
+    #if ($isEdit)
+      &lt;input type="text" class="shortcutPreference" name="$escapedSelectId" id="$escapedSelectId" size="30"
+        data-translatekey="$translationKey"
+        data-defaultvalue="$defaultValue"
+        #if ("$!shortcutPreference" != '')data-shortcutprefnumber='$shortcutPreference.getReference().getObjectNumber()'#end
+        #if ($services.localization.get("XWiki.XWikiUsers_${fieldName}.hint"))
+          #set ($escapedHintId = $escapedSelectId + "_hint")
+          aria-describedby="$escapedHintId"
+        #end
+        #if ($value)
+          value="$value"
+        #elseif ($defaultValue)
+          value="$defaultValue"
+        #end&gt;
+    #else
+      &lt;span&gt;
+        #if("$!value" != '') $!value##
+        #else $defaultValue##
+          &lt;small&gt;($escapetool.xml($services.localization.render('userprofile.default')))&lt;/small&gt;##
```

Comment:
As before, `$defaultValue` should be inserted into the translation with a placeholder so translators can place it freely.

Reply:
Same discussion as https://github.com/xwiki/xwiki-platform/pull/2981#discussion_r1787411826